src/components/ui/RadioButtonGroup/RadioButtonGroup.tsx

```diff
@@ -0,0 +1,59 @@
+import { forwardRef, Fragment, useState } from "react";
+
+import { cn } from "@/utils/helper";
+import radioCheck from "@/assets/icons/radio-check.svg";
+import Icon from "@/components/ui/Icon";
+import { IRadioButtonGroupProps } from "@/components/ui/RadioButtonGroup/RadioButtonGroup.d";
+
+const RadioButtonGroup = forwardRef<HTMLInputElement, IRadioButtonGroupProps>(
+  (props, ref) => {
+    const { radioButtons, ...rest } = props;
+    const [selectedValue, setSelectedValue] = useState(rest.defaultValue);
+
+    return (
+      <Fragment>
+        {radioButtons.map((radioButton, index) => {
```

Comment:
Add edge case checker. At some point, `radioButtons` can be empty or null or undefined. Even if it's protected via typecast, we should still use `?` operator.

Reply:
I have removed this component bhai. Now I'm using single RadioButton component instead of RadioButtonGroup.